src/Rings/mpoly-ideals.jl

```diff
@@ -1968,6 +1998,11 @@ false
   return isempty(lg) || all(iszero, lg)
 end
 
+# In general this depends on the ring, ...
+is_known(::typeof(is_zero), I::MPolyIdeal) = has_attribute(I, :is_zero) || all(is_zero, gens(I))
```

Comment:
The check `all(is_zero, gens(I))` is essentially the entire body of `is_zero(::MPolyIdeal)`. So checking this here amounts to a waste of time...

An implicit rule for `is_known` that we didn't codify but that I think must hold for it to be useful is that `is_known` itself must always be "fast".

Applied here, there are two possibilities:
1. The check `all(is_zero, gens(I))` is considered to be always fast. But then we are better of doing
```suggestion
is_known(::typeof(is_zero), I::MPolyIdeal) = true
```
1. The check `all(is_zero, gens(I))` is not considered to be always fast. But then we should not do it here, and just do
```suggestion
is_known(::typeof(is_zero), I::MPolyIdeal) = has_attribute(I, :is_zero)
```